src/master/filesystem_dump.cc

```diff
@@ -181,14 +185,17 @@ void fs_dumpfree() {
 void xattr_dump() {
 	for (auto i = 0; i < XATTR_DATA_HASH_SIZE; i++) {
 		for (const auto &xattrDataEntry : gMetadata->xattrDataHash[i]) {
-			printf(
-			    "X|i:%10" PRIiNode "|n:%s|v:%s\n", xattrDataEntry.get()->inode,
-			    fsnodes_escape_name(std::string((char *)xattrDataEntry.get()->attributeName.data(),
-			                                    xattrDataEntry.get()->attributeName.size()))
-			        .c_str(),
-			    fsnodes_escape_name(std::string((char *)xattrDataEntry.get()->attributeValue.data(),
-			                                    xattrDataEntry.get()->attributeValue.size()))
-			        .c_str());
+			printf("X|i:%10" PRIiNode "|n:%s|v:%s\n", xattrDataEntry.get()->inode,
+			       gFSOperations->nodeOperations()
+			           ->fsnodes_escape_name(
+			               std::string((char *)xattrDataEntry.get()->attributeName.data(),
```

Comment:
Consider
```suggestion
			              std::string{xattrDataEntry->attributeName.data()}
```
or, if not null terminated,
```suggestion
			              std::string{xattrDataEntry->attributeName.data(), xattrDataEntry->attributeName.size()};
```

Reply:
I will do it in the refactor.